src/server/server.js

```diff
@@ -6,43 +6,168 @@ var fs = require('fs'),
     cordovaServe = require('cordova-serve'),
     send = require('send'),
     url = require('url'),
-    config = require('./config'),
+    Q = require('q'),
     dirs = require('./dirs'),
-    plugins = require('./plugins'),
-    prepare = require('./prepare'),
-    simFiles = require('./sim-files'),
-    simSocket = require('./socket'),
-    log = require('./log');
+    log = require('./utils/log'),
+    SimulationFiles = require('./sim-files'),
+    SocketServer = require('./socket'),
+    pluginSimulationFiles = require('./plugin-files');
 
-var pluginSimulationFiles = require('./plugin-files');
+/**
+ * The simulation server that encapsulates the HTTP server instance and
+ * Web Socket server. It enables to start and stop the server to listening
+ * for new connections.
+ * @param {object} simulator
+ * @constructor
+ */
+function SimulationServer(simulator, project, hostRoot) {
+    this._simulator = simulator;
+    this._project = project;
+    this._hostRoot = hostRoot;
+    this._cordovaServer = cordovaServe();
```

Comment:
I made a small change here, renaming `this._server` to `this._cordovaServer`. It is more representative of the type of instance we are holding.